libs/fluid-elements/tab-group/src/lib/tab/tab.ts

```diff
@@ -0,0 +1,163 @@
+  constructor(public activeTab: string) {
+    super('tabActivated', { bubbles: true, composed: true });
+  }
+}
+
+export class FluidTabDisabledEvent extends CustomEvent<any> {
+  constructor(public disableTab: string) {
+    super('disabled', { bubbles: true, composed: true });
+  }
+}
+
+/**
+ * This is a experimental version of the tab component
+ * It registers itself as `fluid-tab` custom element.
+ */
+export class FluidTab extends LitElement {
+  /** Styles for the tab component */
+  static get styles(): CSSResult {
+    return styles;
+  }
+
+  /**
+   * Defines the tab element with an id attribute
+   * @attr
+   * @type string
+   */
+  @property({ type: String, reflect: true })
+  tabid = `fluid-tab-${_unique++}`;
+
+  /**
+   * Defines whether a tab is disabled or not
+   * @attr
+   * @type boolean
+   */
+  @property({ type: Boolean, reflect: true })
+  get disabled(): boolean {
+    return this._disabled;
+  }
+  set disabled(value: boolean) {
+    const oldValue = this._disabled;
+    if (this._disabled !== value) {
+      this._disabled = value;
+      this.requestUpdate('disabled', oldValue);
+      if (this._disabled) {
+        this.active = false;
+        this.tabindex = -1;
+        this.dispatchEvent(new FluidTabDisabledEvent(this.tabid));
+      }
+    }
+  }
+  private _disabled = false;
+
+  /**
+   * Defines the tabindex attribute
+   * @attr
+   * @type number
+   */
+  @property({ type: Number, reflect: true })
+  get tabindex(): number {
+    return this._tabindex;
+  }
+  set tabindex(value: number) {
+    this._tabindex = value;
+    this.requestUpdate('tabindex');
+  }
+  private _tabindex = 0;
+
+  /**
+   * Defines whether a tab is active or not
+   * @attr
+   * @type boolean
+   */
+  @property({ type: Boolean, reflect: false })
+  get active(): boolean {
+    return this._active;
+  }
+  set active(active: boolean) {
+    // Only set active true if not disabled
+    this._active = this.disabled === false ? active : false;
+    this.requestUpdate('active');
+    if (active) {
+      this.tabindex = 0;
+    } else {
+      this.tabindex = -1;
+    }
+  }
+  private _active = false;
+
+  constructor() {
+    super();
+  }
+
+  /** First updated lifecycle */
+  firstUpdated(props: Map<string | number | symbol, unknown>): void {
+    super.firstUpdated(props);
+  }
+
+  /** Dispatches the custom event  */
+  private dispatchActiveTabEvent(): void {
+    this.dispatchEvent(new FluidTabActivatedEvent(this.tabid));
+  }
+
+  /** Handles the click event */
+  private handleClick(): void {
+    if (!this._active) {
+      this.dispatchActiveTabEvent();
+    }
+  }
+
+  /**
+   * Render function of the custom element. It is called when one of the
+   * observedProperties (annotated with @property) changes.
+   */
+  render(): TemplateResult {
+    const classes = {
+      'fluid-tab': true,
+      'fluid-state--active': this._active,
+    };
+
+    // Linebreak causes the element to have a space
+    return html`<li
```

Comment:
li element is probably not the best one to use here, let's use a `span` or another inline element
```suggestion
    return html`<span
```